Python/Product/PythonTools/PythonTools/WebProject/PythonWebLauncher.cs

```diff
@@ -134,29 +140,41 @@ LaunchConfiguration defaultConfig
         }
 
 
-        private void StartBrowser(string url, Func<bool> shortCircuitPredicate) {
+        private Task StartBrowser(string url, Func<bool> shortCircuitPredicate) {
             Uri uri;
             if (!String.IsNullOrWhiteSpace(url) && Uri.TryCreate(url, UriKind.RelativeOrAbsolute, out uri)) {
+                var tcs = new TaskCompletionSource<object>();
+
                 OnPortOpenedHandler.CreateHandler(
                     uri.Port,
                     shortCircuitPredicate: shortCircuitPredicate,
                     action: () => {
-                        var web = _serviceProvider.GetService(typeof(SVsWebBrowsingService)) as IVsWebBrowsingService;
-                        if (web == null) {
-                            PythonToolsPackage.OpenWebBrowser(url);
-                            return;
+                        try {
```

Comment:
As long as you're touching this could potentially steal NTVS's code for launching the browser: https://github.com/Microsoft/nodejstools/blob/master/Nodejs/Product/Nodejs/Debugger/DebugEngine/AD7Engine.cs#L1089  It ends up being a much nicer experience then just opening a browser on our own because the debugger can coordinate these.